### ISSUE TITLE
Unable to restore custom metric.

### ISSUE THREAD

**Issue by saddamhijazi** [keras-team/keras#15852]
I have written a code for training a neural network in keras-tensorflow where I added a custom metric just for the sake of printing the current learning rate at each epoch (this is useful for me when I use learning schedules). I trained my model and then I tried to load it again for re-training however I got the following error:

Unable to restore custom metric. Please ensure that the layer implements `get_config` and `from_config` when saving. In addition, please use the `custom_objects` arg when calling `load_model()`.

I tried the proposed solution in many other posts in which one use the custom_objects for indicating the custom metric, however I have not been able to solve this issue, below a simple code to reproduce the issue.

```python

import numpy as np
import tensorflow as tf
from tensorflow import keras


print(tf.__version__)

def get_lr_metric(optimizer):
    def lr(y_true, y_pred):
        return optimizer._decayed_lr(tf.float32) # I use ._decayed_lr method instead of .lr
    return lr


_input = tf.keras.layers.Input(shape=(500), name="fbank") # B*T*F*c
out = tf.keras.layers.Dense(50, activation="tanh")(_input)
probabilities = tf.keras.layers.Dense(2, activation="softmax")(out)
model = tf.keras.Model(inputs=_input, outputs=probabilities)
optimizerAdam = keras.optimizers.Adam(learning_rate=1e-5)
lr_metric = get_lr_metric(optimizerAdam)

model.compile(optimizerAdam, loss=tf.keras.losses.CategoricalCrossentropy(), 
              metrics=['accuracy', lr_metric])

model.summary()

x=np.random.rand(300,500)
y=np.random.rand(300,2)
model.fit(x,y,batch_size=100, epochs=2)

path = 'saved_model/'
model.save(path, save_format='tf')

del model
model = tf.keras.models.load_model('saved_model', compile=False,custom_objects={"lr_metric": lr_metric})
print("Done")


```

**Comment by sushreebarsa** [comment]
@sachinprasadhs Was able to reproduce this issue on colab using TF v2.7.0, please find the gist [here](https://colab.sandbox.google.com/gist/sushreebarsa/a2e34b07d25b1b92d00e4cad669aaa04/keras15852.ipynb).Thank you!

**Comment by jvishnuvardhan** [comment]
@saddamhijazi This was resolved in recent `tf-nightly`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/a3204815dd9750a790f29a952e4ede9e/textvectorization.ipynb) is a gist for reference.

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by saddamhijazi** [comment]
Thank you very much, I will close the issue then.

**Comment by jvishnuvardhan** [comment]
Are you satisfied with the resolution of your issue?
[Yes](https://goo.gl/forms/Oe0tEvODFRoI2gJF3)
[No](https://goo.gl/forms/fUjzOfrtkFbrOT8d2)
